include/imex/Dialect/Dist/IR/DistOps.td

```diff
@@ -46,48 +75,149 @@ def Dist_Dialect : Dialect {
 class Dist_Op<string mnemonic, list<Trait> traits = []> :
     Op<Dist_Dialect, mnemonic, traits>;
 
-// Add function prototypes used for calling into distributed runtime
 def RuntimePrototypesOp : Dist_Op<"runtime_prototypes"> {
+    let summary = "Add function prototypes used for calling into distributed runtime";
 }
 
-// Register a ptensor of given shape with a (potentially distributed) runtime.
-// Returns an id to uniquely identify the tensor instance in future interactino with the runtime.
-// The runtime does not own or manage any PTensor memory. When needed by an operation,
-// (local) data needs to be provided.
-def RegisterPTensorOp : Dist_Op<"register_ptensor", []> {
-    // Global shape needed for initial registration. Views are handled by a separate op.
-    let arguments = (ins AnyType: $shape);
+def NProcsOp : Dist_Op<"nprocs", [Pure]> {
+    let summary = "Number of processes for given team";
+    let arguments = (ins AnyType:$team);
+    let results = (outs I64);
+    let builders = [
+        OpBuilder<(ins "::mlir::Value":$team), [{
+            build($_builder, $_state, $_builder.getI64Type(), team);
+        }]>,
+    ];
+}
 
-    // result is an Integer Id
+def PRankOp : Dist_Op<"prank", [Pure]> {
+    let summary = "Process rank in team";
+    let arguments = (ins AnyType:$team);
     let results = (outs I64);
+    let builders = [
+        OpBuilder<(ins "::mlir::Value":$team), [{
+            build($_builder, $_state, $_builder.getI64Type(), team);
+        }]>,
+    ];
 }
 
-// Get the offsets (one for each dimension) of the local partition of a distributed PTensor in number of elements.
-// Partitionings can be N-dimensional but must cut only the first N dimensions.
-def LocalOffsetsOp : Dist_Op<"local_offsets", []> {
-    // Id of tensor as returned by RegisterPTensorOp
-    let arguments = (ins I64Attr: $rank, I64: $ptensor);
+def DistInfoOp : Dist_Op<"distinfo", []> {
+    let summary = "Initialize distributed meta information about a new PTensor.";
+    let description = [{
+        The distributed meta information about a new PTensor provides
+          - the rank of the tensor
+          - the process-local shape
+          - the process-local offsets
+          - the distributed team
+    }];
+    // Global shape needed for initial registration. Views are handled by a separate op.
+    let arguments = (ins I64Attr:$rank, AnyType:$shape, AnyType:$team);
```

Comment:
Why are `$team` and `$shape` `AnyType`?

Reply:
Shapes can be represented in various ways, like as a `shape.shape` or `tensor` (with different element types) or memref.